### PR TITLE
Update astropy data URL in test runner docstring

### PR DESCRIPTION
**Pull request by larrybradley**

### CODE REVIEW

**Comment by bsipocz** [comment]
Interesting, when and why did this URL change?

**Comment by larrybradley** [comment]
🤷 I found it running a linkcheck on `photutils`.

File: astropy/tests/runner.py [hunk 1]
```diff
@@ -463,8 +463,9 @@ def remote_data(self, remote_data, kwargs):
         remote_data : {'none', 'astropy', 'any'}, optional
             Controls whether to run tests marked with @pytest.mark.remote_data. This can be
             set to run no tests with remote data (``none``), only ones that use
-            data from http://data.astropy.org (``astropy``), or all tests that
```
Comment (pllim) [inline astropy/tests/runner.py:466]: Wait a minute... http://data.astropy.org still works.

**Comment by pllim** [comment]
I think this needs to be fixed in `photutils`'s `docs/conf.py`. See

https://github.com/astropy/astropy/blob/986d5b74145c4cf6ee47c303d709832d199df24f/docs/conf.py#L256

File: astropy/tests/runner.py (hunk 1 above)
Comment (larrybradley) [reply astropy/tests/runner.py:466]: Yes, but it redirects to https://www.astropy.org/astropy-data/

Comment (pllim) [reply astropy/tests/runner.py:466]: That might change in the future... 👀

Comment (larrybradley) [reply astropy/tests/runner.py:466]: OK, should I close this then?

Comment (pllim) [reply astropy/tests/runner.py:466]: Yeah... Sorry for the inconvenience!

**Pull request closed without merging**